sdk/profiles/src/main/java/io/opentelemetry/sdk/profiles/data/LocationData.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.profiles.data;
+
+import java.util.List;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Describes a function.
+ *
+ * @see "pprofextended.proto::Location"
+ */
+@Immutable
+public interface LocationData {
+
+  /**
+   * Unique nonzero id for the location. Could use instruction addresses or any integer sequence as
+   * ids.
+   *
+   * @deprecated retained only for pprof compatibility.
+   */
+  @Deprecated
+  long getId();
+
+  /**
+   * The index of the corresponding profile.Mapping for this location. It can be unset if the
+   * mapping is unknown or not applicable for this profile type.
+   */
+  long getMappingIndex();
+
+  /** The instruction address for this location, if available. */
+  long getAddress();
+
+  /**
+   * Multiple line indicates this location has inlined functions, where the last entry represents
+   * the caller into which the preceding entries were inlined.
+   */
+  List<LineData> getLines();
+
+  /**
+   * Provides an indication that multiple symbols map to this location's address, for example due to
+   * identical code folding by the linker.
+   */
+  boolean isFolded();
+
+  /** Type of frame (e.g. kernel, native, python, hotspot, php). Index into string table. */
+  int getTypeIndex();
```

Comment:
should this be an enum?

Reply:
Eventually perhaps, but that would require some work on the semantic conventions first, to define the valid values. At this stage it's not clear what they are and I'd have little confidence we could reliably identify them all without widespread real world usage first. Besides, the encoding of an enum field and a pointer into a string table are different on the wire, even if the API allows only the same set of string values as the enum. So for use in the Marshalers, which is what this API is really for, keeping it as an int is easier.

Sidenote: there are points in the API where the data type for indexes is inconsistent. The proto msotly allows very large arrays and uses 64 bit indexes, whilst Java arrays are 32 bit indexed. My first cut used 32 bit for all index fields and thus could not be used to encode all possible valid values of the proto, whilst still being adequate for all realistic usage. The current version mostly uses long for indexes despite the space overhead. This field is unusual in actually being a 32 bit in the underlying proto.

Reply:
Can't do enums yet, but work is underway to define common values, which may then become String constants and perhaps eventually enums. Can revisit the interface when that spec work is further along.
https://github.com/open-telemetry/opentelemetry-proto/issues/555